lib/services/clusterconfig.go

```diff
@@ -209,3 +197,60 @@ func MarshalClusterConfig(clusterConfig ClusterConfig, opts ...MarshalOption) ([
 		return nil, trace.BadParameter("unrecognized cluster config version %T", clusterConfig)
 	}
 }
+
+// GetClusterConfigOverrideSchema returns the schema with optionally injected
+// schema for extensions.
+func GetClusterConfigOverrideSchema(extensionSchema string) string {
+	var clusterConfigOverrideSchema string
+	if clusterConfigOverrideSchema == "" {
+		clusterConfigOverrideSchema = fmt.Sprintf(ClusterConfigOverrideSchemaTemplate, "")
+	} else {
+		clusterConfigOverrideSchema = fmt.Sprintf(ClusterConfigOverrideSchemaTemplate, ","+extensionSchema)
+	}
+	return fmt.Sprintf(V2SchemaTemplate, MetadataSchema, clusterConfigOverrideSchema, DefaultDefinitions)
+}
+
+// UnmarshalClusterConfigOverride unmarshals the ClusterConfigOverride resource from JSON.
+func UnmarshalClusterConfigOverride(bytes []byte, opts ...MarshalOption) (types.ClusterConfigOverride, error) {
+	var override types.ClusterConfigOverrideV3
+	schema := GetClusterConfigOverrideSchema("")
+	if err := unmarshalResource(schema, &override, bytes, opts...); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if err := override.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &override, nil
+}
+
+// TODO(andrej): Use this helper for all services.Unarmshal* functions.
+// Also make CheckAndSetDefaults part of Resource interface.
+func unmarshalResource(schema string, res Resource, bytes []byte, opts ...MarshalOption) error {
+	if len(bytes) == 0 {
+		return trace.BadParameter("missing resource data")
+	}
+
+	cfg, err := CollectOptions(opts)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if cfg.SkipValidation {
+		if err := utils.FastUnmarshal(bytes, res); err != nil {
+			return trace.BadParameter(err.Error())
+		}
+	} else {
+		err = utils.UnmarshalWithSchema(schema, res, bytes)
```

Comment:
Nit: Let's be consistent and either use `if err := xxx; err != nil` or this style, esp. when they're so close together.